lib/Data/Filesystem.php

```diff
@@ -358,24 +364,26 @@ protected function _getExpiredPastes($batchsize)
     {
         $pastes = array();
         $count  = 0;
+        $opened = 0;
+        $limit  = $batchsize * 10; // try at most 10 times $batchsize pastes before giving up
         $time   = time();
-        foreach ($this->_getPasteIterator() as $file) {
-            if ($file->isDir()) {
-                continue;
-            }
-            $pasteid = $file->getBasename('.php');
+        $files  = $this->getAllPastes();
+        shuffle($files);
```

Comment:
Neither was array_rand, that was used originally. The point of the randomization was to ensure that the FS storage tests different pastes every time it is called. The purge in the FS storage can't use metadata, other then reading the paste fully, so it will end up reading (potentially many) not yet expired pastes until it can find expired ones to be purged. If we always start at the beginning, we will eventually have to read through all of the pastes only to find nothing left to purge. The shuffle is there to increase the chances that it will find expired pastes from the end of the list - it does not have to be secure, just needs to return pastes in different order every time.

Reply:
Is "always read from the end" an option or possible, though I guess even then shuffle is better anyway.